modules/http/src/main/java/org/glassfish/grizzly/http/HttpServerFilter.java

```diff
@@ -1195,6 +1204,24 @@ private void sendBadRequestResponse(final FilterChainContext ctx,
         commitAndCloseAsError(ctx, response);
     }
 
+    private void sendEntityTooLargeResponse(final FilterChainContext ctx,
+                                        final HttpResponsePacket response) {
+        if (response.getHttpStatus().getStatusCode() < 400) {
```

Comment:
Are we sure that if the error code is 3xx or 2xx is always because the request entity is too large?

Reply:
I think the if is there because if the response had another error(above 400) it has to answer with this one and not replace it with the new one.